api/catalog/api/models/media.py

```diff
@@ -242,7 +247,51 @@ def save(self, *args, **kwargs):
             same_reports.update(status=self.status)
 
 
-class AbstractDeletedMedia(OpenLedgerModel):
+class PerformIndexUpdateMixin:
+    @property
+    def indexes(self):
+        return [self.es_index, f"{self.es_index}-filtered"]
+
+    def _perform_index_update(self, method: str, raise_errors: bool, **es_method_args):
+        """
+        Call ``method`` on the Elasticsearch client.
+
+        Automatically handles ``DoesNotExist`` warnings, forces a refresh,
+        and calls the method for origin and filtered indexes.
+        """
+        logger = parent_logger.getChild("PerformIndexUpdateMixin._perform_index_update")
+        es: Elasticsearch = settings.ES
+
+        try:
+            document_id = self.media_obj.id
+        except self.media_class.DoesNotExist:
+            if raise_errors:
+                raise ValidationError(
+                    f"No '{self.media_class.__name__}' instance"
+                    f"with identifier {self.media_obj.identifier}."
+                )
+
+        for index in self.indexes:
+            try:
+                getattr(es, method)(
+                    index=index,
+                    id=document_id,
+                    refresh=True,
+                    **es_method_args,
+                )
+            except TransportError as e:
+                if e.status_code == 404:
+                    # This is expected for the filtered index, but we should still
+                    # log, just in case.
+                    logger.warning(
```

Comment:
If this is expected should it be at debug level then?
```suggestion
                    logger.debug(
```

Reply:
It's only expected for the filtered index and only then when the filtered index excluded it. I could refine this a bit to register an issue with Sentry when the 404 occurred on the origin index (documents shouldn't be missing there), but I'd still like to have a visible, historically accessible log (i.e., one we don't need to turn on to have appear in the logs) so we can investigate if documents aren't being excluded.

This code does not run often, so this new warning log isn't going to cause an big or even noticeable increase in the number of logs stored.